File: saml/src/main/java/com/zorroa/security/saml/ZorroaMetadataFilter.java
package com.zorroa.security.saml;

import org.springframework.security.saml.metadata.MetadataGenerator;
import org.springframework.security.saml.metadata.MetadataGeneratorFilter;

import javax.servlet.http.HttpServletRequest;

public class ZorroaMetadataFilter extends MetadataGeneratorFilter {

    private final String baseUrl;

    public ZorroaMetadataFilter(String baseUrl, MetadataGenerator generator) {
        super(generator);
        this.baseUrl = baseUrl;
    }

    @Override
    protected String getDefaultBaseURL(HttpServletRequest request) {
        return baseUrl;
    }
}

File: saml/src/main/kotlin/com/zorroa/security/saml/ZorroaMetadataFilter.kt
package com.zorroa.security.saml

import org.springframework.security.saml.metadata.MetadataGenerator
import org.springframework.security.saml.metadata.MetadataGeneratorFilter

import javax.servlet.http.HttpServletRequest

class ZorroaMetadataFilter(private val baseUrl: String, generator: MetadataGenerator) : MetadataGeneratorFilter(generator) {

    override fun getDefaultBaseURL(request: HttpServletRequest): String {
        return baseUrl
    }
}
